test/common/memory/memory_release_test.cc

```diff
@@ -129,9 +128,9 @@ TEST_F(MemoryReleaseTest, ReleaseRateAboveZeroCustomIntervalMemoryReleased) {
   step(std::chrono::milliseconds(2000));
   b.reset();
   step(std::chrono::milliseconds(2000));
-  EXPECT_EQ(2UL, stats_.counter("memory_release_test.tcmalloc.released_by_timer").value());
```

Comment:
@KBaichoo think i figured out the flakiness in past version of tests. While spending time trying to understand the internals of tcmalloc and blaming tcmalloc, the root cause was in how i verified the stats. Switched to `TestUtility::waitForCounterEq` instead of instant comparison and it fixed the occasional flakiness.